ux.c

```diff
@@ -38,12 +42,23 @@ static void ux_clear_line(void)
 
 void ux_init(void)
 {
-	struct winsize w;
+#if WIN32
+    CONSOLE_SCREEN_BUFFER_INFO csbi;
+	int columns;
+
+    HANDLE stdoutHandle = GetStdHandle(STD_OUTPUT_HANDLE);
+    if (GetConsoleScreenBufferInfo(stdoutHandle, &csbi)) {
+		columns = csbi.srWindow.Right - csbi.srWindow.Left + 1;
+		ux_width = MIN(columns, UX_PROGRESS_SIZE_MAX);
+    }
+#else
 	int ret;
+	struct winsize w;
 
 	ret = ioctl(STDOUT_FILENO, TIOCGWINSZ, &w);
 	if (!ret)
 		ux_width = MIN(w.ws_col, UX_PROGRESS_SIZE_MAX);
+#endif
 }
```

Comment:
Please put `#if` outside of the function, basically having one `ux_init()` for Win32 and one for all other platforms.